src/biokbase/narrative/jobs/job.py

```diff
@@ -341,16 +330,18 @@ def _update_state(self, state: dict) -> None:
         # Check if there would be no change in updating
         # i.e., if state <= self._acc_state
         if self._acc_state is not None:
-            if {**self._acc_state, **state} == self._acc_state:
+            if merge(self._acc_state, state) == self._acc_state:
```

Comment:
why merge instead of just checking whether `{**self._acc_state, **state}` would be the same as `self._acc_state`?